test/unit/v35.test.js

```diff
@@ -108,6 +108,29 @@ describe('v5', () => {
     assert.ok(v3('hello.example.com', '00000000-0000-0000-0000-000000000000'));
   });
 
+  test('v3 parsing non RFC uuid values', () => {
+    assert.strictEqual(
+      v3(
+        'hello.example.com',
+        // equal '00000000-0000-0000-0000-000000000000'
+        '00000000000000000000000000000000',
```

Comment:
@ctavan I tried to do it in another PR that you rejected: https://github.com/uuidjs/uuid/pull/439
=)

Now I just want to prepare a module for the next v35 optimization (and not disrupt current behavior).

Reply:
Uhm well, we rejected #439 because of [the complexity of the UUID parsing code](https://github.com/uuidjs/uuid/pull/439#issuecomment-627442855).

That doesn't mean that an additional input validation of the UUID v3/v5 namespace parameter would make sense. I'd be happy to accept such a change if it was presented in a non-complex manner!

The thing about the test cases that you are adding here is that they actually reveal _unintended_ behavior, however I believe the test-cases should only document the _intended_ behavior!

So in fact you are just uncovering a flaw in the library and we should rather fix that flaw and turn these test cases into `assert.throws()`.